client/cpp/kserve-api/samples/grpc_infer_resnet.cpp

```diff
@@ -0,0 +1,175 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <fstream>
+#include <iostream>
+#include <iterator>
+#include <vector>
+#include <string>
+
+#include <cxxopts.hpp>
+
+#include "grpc_client.h"
+
+namespace tc = triton::client;
+
+#define FAIL_IF_ERR(X, MSG)                                              \
+    {                                                                    \
+        tc::Error err = (X);                                             \
+        if (!err.IsOk()) {                                               \
+            std::cerr << "error: " << (MSG) << ": " << err << std::endl; \
+            exit(1);                                                     \
+        }                                                                \
+    }
+
+std::vector<uint8_t> Load(std::string fileName) {
+    std::ifstream fileImg(fileName, std::ios::binary);
+    fileImg.seekg(0, std::ios::end);
+    int bufferLength = fileImg.tellg();
+    fileImg.seekg(0, std::ios::beg);
+
+    char* buffer = new char[bufferLength];
+    fileImg.read(buffer, bufferLength);
+
+    return std::vector<uint8_t>(buffer, buffer + bufferLength);
+}
+
+int main(int argc, char** argv) {
+    cxxopts::Options opt("grpc_infer_resnet", "Sends requests via KServe gRPC API.");
+
+    // clang-format off
+    opt.add_options()
+    ("h,help", "Show this help message and exit")
+    ("images_list", "Path to a file with a list of labeled images. ", cxxopts::value<std::string>(), "IMAGES")
+    ("labels_list", "Path to a file with a list of labels. ", cxxopts::value<std::string>(), "LABELS")
+    ("grpc_address", "Specify url to grpc service. ", cxxopts::value<std::string>()->default_value("localhost"), "GRPC_ADDRESS")
+    ("grpc_port", "Specify port to grpc service. ", cxxopts::value<std::string>()->default_value("9000"), "PORT")
+    ("input_name", "Specify input tensor name. ", cxxopts::value<std::string>()->default_value("0"), "INPUT_NAME")
+    ("output_name", "Specify input tensor name. ", cxxopts::value<std::string>()->default_value("1463"), "OUTPUT_NAME")
+    ("model_name", "Define model name, must be same as is in service. ", cxxopts::value<std::string>()->default_value("resnet"), "MODEL_NAME")
+    ("model_version", "Define model version.", cxxopts::value<std::string>(), "MODEL_VERSION")
+    ("timeout", "Request timeout.", cxxopts::value<int>()->default_value("0"), "TIMEOUT")
+    ;
+    // clang-format on
+
+    auto args = opt.parse(argc, argv);
```

Comment:
What if user passes string to timeout parameter? Does cxxopts perform input validation? If so, does cxxopts do it via exceptions? If so, I do not see we catch it or handle it